File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;

        // 주어진 input을 -를 기준으로 분리
        String[] input = br.readLine().split("-");

        for(int i=0; i < input.length; i++) {
            // -를 기준으로 분리한 것들 중 + 연산이 들어있으면 sum처리
            String[] divided = input[i].split("\\+");
            int[] tempArray = Arrays.stream(divided).mapToInt(Integer::parseInt).toArray();
            input[i] = String.valueOf(Arrays.stream(tempArray).sum());
        }

        br.close();

        int result = 0;
        for(int i = 0; i < input.length; i++) {
            if(i == 0) {
                result += Integer.parseInt(input[i]);
            }
            else {
                result -= Integer.parseInt(input[i]);
            }
        }
        bw.write(String.valueOf(result));
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys

input = sys.stdin.readline

# -를 기준으로 나눈다음
arr = input().split("-")
result = 0

# 첫 번째 나눈값을 모두 더하고
for num in arr[0].split("+"):
    result += int(num)
    
# 나머지 값들을 모두 뺀다
for i in arr[1:]:
    for j in i.split("+"):
        result -= int(j)

print(result)